src/main/java/uk/gov/justice/probation/courtcasematcher/event/EventListener.java

```diff
@@ -90,20 +89,18 @@ public void courtCaseMatchEvent(CourtCaseMatchEvent courtCaseEvent) {
                             .orElseGet(() -> NameHelper.getNameFromFields(courtCase.getDefendantName()));
 
         matcherService.getSearchResponse(name, courtCase.getDefendantDob(), courtCase.getCourtCode(), courtCase.getCaseNo())
-            .switchIfEmpty(Mono.defer(() -> Mono.just(SearchResponse.builder()
-                                                            .matchedBy(OffenderSearchMatchType.NOTHING)
-                                                            .matches(Collections.emptyList())
-                                                            .build())))
+            .doOnError(throwable -> {
+                telemetryService.trackOffenderMatchFailureEvent(courtCase);
+                courtCaseService.createCase(courtCase, SearchResponse.builder()
```

Comment:
I noticed the `createCase()` call here is duplicated at line 101 so I was curious if there was a way to achieve the same thing but without that duplication. After a bit of playing around I managed to come up with this: 
```
        matcherService.getSearchResponse(name, courtCase.getDefendantDob(), courtCase.getCourtCode(), courtCase.getCaseNo())
            .doOnSuccess(searchResponse -> telemetryService.trackOffenderMatchEvent(courtCase, searchResponse))
            .doOnError(throwable -> telemetryService.trackOffenderMatchFailureEvent(courtCase))
            .onErrorResume(throwable -> Mono.just(SearchResponse.builder()
                                .matchedBy(OffenderSearchMatchType.NOTHING)
                                .matches(Collections.emptyList())
                                .build()))
            .subscribe(searchResponse -> courtCaseService.createCase(courtCase, searchResponse))
```

It reads quite differently but quite well I think, I like the fact that it's able to use pure Lambda's without any need for `{}` blocks. What do you think?

Reply:
Yes - will replace. There weren't any blocks() there in any case but yes - there's less duplication and it reads a bit better.

Reply:
sorry I meant a 'block' of code with a return statement enclosed in curly brackets as opposed to a lambda expression - not quite sure what the right terminology is to be honest but:
```
foo((f) -> {
    // do multiple things
    // ...
    return f.baz()
})
```
rather than:
```
foo((f) -> f.bar())
```